pkg/kv/kvserver/rditer/select.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package rditer
+
+import "github.com/cockroachdb/cockroach/pkg/roachpb"
+
+type StateMachineSelectionOptions struct {
```

Comment:
Do you plan to use this struct in isolation from `SelectionOptions`? If not, I would suggest inlining it, to avoid the verbosity that it carries (long name repetitions + extra nesting on every `Select` call; e.g. see calls in `replica_raftstorage.go`).

Reply:
Good idea, this was unnecessarily verbose. Done.